app/models/list.rb

```diff
@@ -39,4 +39,12 @@ def add_item params
   def is_shared_with?(u)
     user != u
   end
+
+  def shared_list_owner
+    user.first_name + " " + user.last_name
+  end
+
+  def number_shared_with
+    SharedList.where(list_id: self.id).length
```

Comment:
So here's a fun bit of performance trivia: calling `length` here will cause Rails to load all the SharedList records that fit the SQL conditions as an array, then count the array.  On a small enough data set, that wouldn't be a big deal, but on a larger dataset it would be problematic.  Since you only want the number of records rather than the data, use [`count`](https://api.rubyonrails.org/classes/ActiveRecord/Calculations.html#method-i-count) which translates to a `COUNT()` SQL call and prevent retrieving data and initializing objects.

```suggestion
    SharedList.where(list_id: self.id).count
```